includes/discount-functions.php

```diff
@@ -967,170 +945,489 @@ function edd_apply_preset_discount() {
 }
 add_action( 'init', 'edd_apply_preset_discount', 999 );
 
+
 /**
- * Updates discounts that are expired or at max use (that are not already marked as so) as inactive or expired
+ * Backwards compatibility filters for get_post_meta() calls on discount codes.
  *
- * @since 2.6
- * @return void
-*/
-function edd_discount_status_cleanup() {
+ * @since  3.0
+ *
+ * @param int    $object_id The object ID post meta was requested for.
+ * @param string $meta_key  The meta key requested.
+ * @param bool   $single    If the person wants the single value or an array of the value.
+ * @return mixed The value to return.
+ */
+function _edd_discount_post_meta_bc_filter( $value, $object_id, $meta_key, $single ) {
 	global $wpdb;
 
-	// We only want to get 25 active discounts to check their status per step here
-	$cron_discount_number   = apply_filters( 'edd_discount_status_cleanup_count', 25 );
-	$discount_ids_to_update = array();
-	$needs_inactive_meta    = array();
-	$needs_expired_meta     = array();
-
-	// start by getting the last 25 that hit their maximum usage
-	$args = array(
-		'suppress_filters' => false,
-		'post_status'      => array( 'active' ),
-		'posts_per_page'   => $cron_discount_number,
-		'order'            => 'ASC',
-		'meta_query'       => array(
-			'relation' => 'AND',
-			array(
-				'key'     => '_edd_discount_uses',
-				'value'   => 'mt1.meta_value',
-				'compare' => '>=',
-				'type'    => 'NUMERIC',
-			),
-			array(
-				'key'     => '_edd_discount_max_uses',
-				'value'   => array( '', 0 ),
-				'compare' => 'NOT IN',
-			),
-			array(
-				'key'     => '_edd_discount_max_uses',
-				'compare' => 'EXISTS',
-			),
-		),
-	);
+	$meta_keys = apply_filters( 'edd_post_meta_discount_backwards_compat_keys', array(
+		'_edd_discount_status',
+		'_edd_discount_amount',
+		'_edd_discount_uses',
+		'_edd_discount_name',
+		'_edd_discount_code',
+		'_edd_discount_expiration',
+		'_edd_discount_start',
```

Comment:
This is missing a few keys: `_edd_discount_is_single_use`, `_edd_discount_is_not_global`, `_edd_discount_product_condition`, _edd_discount_min_price`, `_edd_discount_max_uses`.

Same thing at line 998.

Reply:
And 1052.

Reply:
And 1094